mythril/laser/ethereum/instructions.py

```diff
@@ -1063,28 +1079,101 @@ def codecopy_(self, global_state: GlobalState) -> List[GlobalState]:
             global_state.mstate.stack.pop(),
             global_state.mstate.stack.pop(),
         )
+        code = global_state.environment.code.bytecode
+        if code[0:2] == "0x":
+            code = code[2:]
+        code_size = len(code) // 2
 
-        if (
-            isinstance(global_state.current_transaction, ContractCreationTransaction)
-            and code_offset >= len(global_state.environment.code.bytecode) // 2
-        ):
+        if isinstance(global_state.current_transaction, ContractCreationTransaction):
             # Treat creation code after the expected disassembly as calldata.
             # This is a slightly hacky way to ensure that symbolic constructor
             # arguments work correctly.
-            offset = code_offset - len(global_state.environment.code.bytecode) // 2
+            offset = code_offset - code_size
             log.warning("Doing hacky thing offset: {} size: {}".format(offset, size))
-            return self._calldata_copy_helper(
-                global_state, global_state.mstate, memory_offset, offset, size
-            )
-        else:
-            return self._code_copy_helper(
-                code=global_state.environment.code.bytecode,
-                memory_offset=memory_offset,
-                code_offset=code_offset,
-                size=size,
-                op="CODECOPY",
-                global_state=global_state,
-            )
+
+            if isinstance(global_state.environment.calldata, SymbolicCalldata):
```

Comment:
With this, in theory, we should no longer need this logic in `_code_copy_helper`:
```
        if concrete_size == 0 and isinstance(
            global_state.current_transaction, ContractCreationTransaction
        ):
            if concrete_code_offset >= len(code) // 2:
                Instruction._handle_symbolic_args(global_state, concrete_memory_offset)
                return [global_state]
```
And `_handle_symbolic_args` can be deleted.

That was our previous, hacky solution that didn't work in a couple situations.